Fixes for Payara mnaged profile

OndrejM opened the pull request:
I identified 3 issues when running `mvn clean install -Ppayara-build-managed-4`:
- missing configuration for Payara in data module projects
- wrong assertion in Java8Test test (correct me if I'm wrong, but the lists should contain the same elements, but not necessarily in the same order)
- failing tests in ContainerCtrlTckTest

I'm sending PR for the 1st and 2nd issue. The 3rd issue remains - it seems that there is a problem with Jersey integration. Jersey cannot find a context for JNDI lookup, I will need to investigate further what is the reason. The following commit would fi the issue, but it effectively disables all the tests, as Payara Server 162 uses Weld 2.3.2.Final, which is out of specified range: https://github.com/apache/deltaspike/commit/8166155a3cbdd039b0a2b4bf2a95db3da008e1bf

johnament commented on deltaspike/modules/data/test-java8/src/test/java/org/apache/deltaspike/data/test/java8/test/Java8Test.java:
```diff
@@ -19,6 +19,7 @@
 
 package org.apache.deltaspike.data.test.java8.test;
 
+import java.util.*;
```
Please correct the wildcard import.

johnament commented:
Can you squash the commits?

johnament commented:
Also please reference the JIRA ticket https://issues.apache.org/jira/browse/DELTASPIKE-1181

OndrejM commented:
I will squash the commits. I just wanted to separate fixes for multiple issues - every commit fixes a single issue.

OndrejM commented:
The commits are squashed now. I also fixed a compilation issue - the Java8 test project did not specify Java8 compilation target, so it took 1.6 from a parent POM.

I'm not able to add reference to DELTASPIKE-1181 - I changed the commit message in my local repo, but force push does not update the commit message.

OndrejM commented:
Now the build is failing in Jenkins on some weird issue with mvn. The compilation error is gone though.
